provider/infoblox.go

```diff
@@ -0,0 +1,483 @@
+			}
+
+			logrus.Infof(
+				"Creating %s record named '%s' to '%s' for Infoblox DNS zone '%s'.",
+				ep.RecordType,
+				ep.DNSName,
+				ep.Target,
+				zone,
+			)
+
+			mapObjRes, err := p.mapObjRes(ep, false)
+			if err != nil {
+				logrus.Errorf(
+					"Failed to retrieve %s record named '%s' to '%s' for DNS zone '%s': %v",
+					ep.RecordType,
+					ep.DNSName,
+					ep.Target,
+					zone,
+					err,
+				)
+				continue
+			}
+			_, err = p.client.CreateObject(mapObjRes.obj)
+			if err != nil {
+				logrus.Errorf(
+					"Failed to create %s record named '%s' to '%s' for DNS zone '%s': %v",
+					ep.RecordType,
+					ep.DNSName,
+					ep.Target,
+					zone,
+					err,
+				)
+			}
+		}
+	}
+}
+
+func (p *InfobloxProvider) deleteRecords(deleted infobloxChangeMap) {
+	// Delete records first
+	for zone, endpoints := range deleted {
+		for _, ep := range endpoints {
+			if p.dryRun {
+				logrus.Infof("Would delete %s record named '%s' for Infoblox DNS zone '%s'.", ep.RecordType, ep.DNSName, zone)
+			} else {
+				logrus.Infof("Deleting %s record named '%s' for Infoblox DNS zone '%s'.", ep.RecordType, ep.DNSName, zone)
+				mapObjRes, err := p.mapObjRes(ep, true)
+				if err != nil {
+					logrus.Errorf(
+						"Failed to retrieve %s record named '%s' to '%s' for DNS zone '%s': %v",
+						ep.RecordType,
+						ep.DNSName,
+						ep.Target,
+						zone,
+						err,
+					)
+					continue
+				}
+				switch ep.RecordType {
+				case endpoint.RecordTypeA:
+					for _, record := range *mapObjRes.res.(*[]ibclient.RecordA) {
+						_, err = p.client.DeleteObject(record.Ref)
+					}
+				case endpoint.RecordTypeCNAME:
+					for _, record := range *mapObjRes.res.(*[]ibclient.RecordCNAME) {
+						_, err = p.client.DeleteObject(record.Ref)
+					}
+				case endpoint.RecordTypeTXT:
+					for _, record := range *mapObjRes.res.(*[]ibclient.RecordTXT) {
+						_, err = p.client.DeleteObject(record.Ref)
+					}
+				}
+				if err != nil {
+					logrus.Errorf(
+						"Failed to delete %s record named '%s' for Infoblox DNS zone '%s': %v",
+						ep.RecordType,
+						ep.DNSName,
+						zone,
+						err,
+					)
+				}
+			}
+		}
+	}
+}
+
+func (p *InfobloxProvider) updateRecords(updated infobloxChangeMap) {
+	for zone, endpoints := range updated {
+		for _, ep := range endpoints {
+			if p.dryRun {
+				logrus.Infof(
+					"Would update %s record named '%s' to '%s' for Infoblox DNS zone '%s'.",
+					ep.RecordType,
+					ep.DNSName,
+					ep.Target,
+					zone,
+				)
+				continue
+			}
+
+			logrus.Infof(
+				"Updating %s record named '%s' to '%s' for Infoblox DNS zone '%s'.",
+				ep.RecordType,
+				ep.DNSName,
+				ep.Target,
+				zone,
+			)
+
+			mapObjRes, err := p.mapObjRes(ep, true)
+			if err != nil {
+				logrus.Errorf(
+					"Failed to retrieve %s record named '%s' to '%s' for DNS zone '%s': %v",
+					ep.RecordType,
+					ep.DNSName,
+					ep.Target,
+					zone,
+					err,
+				)
+				continue
+			}
+			switch ep.RecordType {
```

Comment:
This switch block should also be a method on top of mapObjRes.

Reply:
I have been trying to figure out some uniform way of dealing with the interface conversion to cut down on the repetition, but I can honestly say that I don't have enough go chops to claim to know where to start. Reflection came to mind, but what I read discouraged using reflection unless absolutely necessary. Mercy?